### DISCUSSION TITLE
The expression 'x => x.Email.Address' is not a valid member access expression .

### DISCUSSION THREAD

**Discussion by cmssantos** [balta-io/forum#13] [Tira Dúvidas] [answered]
Boa noite,

Estou usando EF6. A minha entidade User usa um tipo complexo "Email", com a propriedade "Addres". Na hora de criar o migration apresenta o erro abaixo:

The expression 'x => x.Email.Address' is not a valid member access expression. The expression should represent a simple property or field access: 't => t.MyProperty'. (Parameter 'memberAccessExpression')

public class Email : ValueObject
{
    protected Email () { }

    public Email(string address)
    {
        Address = address;
    }

    public string Address { get; private set; }
}

**Comment by andrebaltieri** [comment, marked as answer]
@cmssantos bom dia! 

No caso, no seu mapeamento você precisa especificar isto:
```csharp
builder.OwnsOne(x => x.Name)
            .Property(x => x.FirstName)
            .HasColumnName("FirstName")
            .IsRequired(true)
            .HasMaxLength(60)
            .HasColumnType("NVARCHAR");
```

**Reply by cmssantos** [reply]
Bom dia @andrebaltieri.

Obrigado pelo rápido retorno. Porém agora tem outro problema:

"Unable to determine the relationship represented by navigation 'Email.Notifications' of type 'IReadOnlyCollection<Notification>'. Either manually configure the relationship, or ignore this property using the '[NotMapped]' attribute or by using 'EntityTypeBuilder.Ignore' in 'OnModelCreating'."

Está entendento eu o Notifiable<Notifications> está gerando um relacionamento com uma entidade "Notifications".

Vi que antes herdava apenas do Notifiable, agora é Notifiable<Notifications>. Tanto a entidade User, quando o ValueObject Email, herdam..

Assim recolve para Email 

    builder.Ignore(x => x.Email)
            .Property(x => x.Notifications);

Porém agora reclamo do User.Notifications. Neste caso não sei como acessar.

The property 'User.Notifications' is of type 'IReadOnlyCollection<Notification>' which is not supported by the current database provider. Either change the property CLR type, or ignore the property using the '[NotMapped]' attribute or by using 'EntityTypeBuilder.Ignore' in 'OnModelCreating'.



Poseria me auxiliar?

Grato.

**Reply by andrebaltieri** [reply]
@cmssantos qual seria o curso/aula que este erro ocorre?
